File: crates/domain-scan-core/tests/fixtures/java/interfaces.java
package com.example;

public interface UserRepository {
    User findById(String id);
    List<User> findAll();
    void save(User user);
    void delete(String id);
}

interface Closeable {
    void close();
}

public interface EventHandler<T> extends Closeable {
    void handle(T event);
    boolean canHandle(String eventType);
}

interface ReadOnly {
    Object get(String key);
}

File: crates/domain-scan-core/tests/fixtures/kotlin/interfaces.kt
package com.example

interface UserRepository {
    fun findById(id: String): User?
    fun findAll(): List<User>
    fun save(user: User)
    fun delete(id: String)
}

interface Closeable {
    fun close()
}

interface EventHandler<T> : Closeable {
    fun handle(event: T)
    fun canHandle(eventType: String): Boolean
}

private interface InternalCache {
    fun get(key: String): Any?
}
